wire-ios/WireCommonComponents/Analytics/WireAnalytics.swift

```diff
@@ -16,5 +16,31 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
+import WireSystem
+
 /// Namespace for analytics tools.
-public enum WireAnalytics {}
+public enum WireAnalytics {
+
+    private static var isSetUp = false
+
+    public static func setup() {
+        guard !isSetUp else {
```

Comment:
I think we need to have some kind of lock here. I think we had issues without it in the notification service right @netbe ?

Reply:
yes the NSE can be reused multiple times, we should guarantee that the logger is added only once

Reply:
so it should be thread safe too

Reply:
I would keep the lock @caldrian unless there's something I am missing.

Reply:
Thanks, added a lock.